test/parallel/test-repl-require.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+
+process.chdir(common.fixturesDir);
+const repl = require('repl');
+
+const server = net.createServer(conn => {
+  repl.start('', conn).on('exit', () => {
+    conn.destroy();
+    server.close();
+  });
+});
+
+const host = common.localhostIPv4;
+const port = common.PORT;
+const options = { host, port };
+
+server.listen(options, function() {
+  const conn = net.connect(options);
+  conn.setEncoding('utf8');
+  conn.on('data', data => answer += data);
+  conn.write('require("baz")\n.exit\n');
+});
+
+var answer = '';
```

Comment:
Tiny nit: while `answer` gets hoisted, I'd be more clear if you move this above the `server.listen` block.